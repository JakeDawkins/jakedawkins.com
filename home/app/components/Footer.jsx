var React = require('react');
var PropTypes = React.PropTypes;
let {Grid, Row, Col} = require('react-bootstrap');

var Footer = React.createClass({

    render: function() {
        return (
            <footer>
                <Grid>
                    <Col md={8} mdOffset={2} sm={12} xs={12} className="hard">
                        <hr />
                        <p className="text-muted text-center">&copy;2016 Jake Dawkins</p>
                    </Col>
                </Grid>
            </footer>
        );
    }

});

module.exports = Footer;
