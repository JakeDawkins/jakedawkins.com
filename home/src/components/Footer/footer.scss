@import "../../styles/variables";

//@extend-elements
//original selectors
//footer p, footer a
%extend_1 {
  text-decoration: none;
  font-family: $font--header;
  font-weight: 300;
  font-style: normal;
  color: $color--white;
  text-align: center;
  margin: 0;
}

footer {
  background: $color--blue;
  padding: 1.563rem 0;
  margin-top: 100px;
  p {
    @extend %extend_1;
  }
  a {
    @extend %extend_1;
    &:hover {
      color: $color--red;
    }
  }
}
