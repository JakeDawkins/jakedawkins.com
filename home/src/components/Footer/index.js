import React, { PropTypes } from "react";
import "./footer.scss";

const Footer = (props) =>
  <footer style={props.collapseMargins ? {marginTop: 0} : null}>
    <p>&copy;2016 Jake Dawkins. Based off theme by <a href="https://willianjusten.com.br">Willian Justen</a></p>
  </footer>

Footer.propTypes = {
    collapseMargins: PropTypes.bool,
}

export default Footer;
