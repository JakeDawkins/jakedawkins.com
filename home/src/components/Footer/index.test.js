import Footer from './';
import React from 'react';
import renderer from 'react-test-renderer';

describe("Footer", () => {

  it("should exist", () => {
    const footer = renderer.create(
      <Footer />
    );

    expect(footer).toMatchSnapshot();
  });

  it("should render props correctly", () => {
    // const footerNoCollapse = renderer.create(
    //   <Footer />
    // );

    const footerWithCollapse = renderer.create(
      <Footer
        collapseMargins={true}
      />
    );

    expect(footerWithCollapse).toMatchSnapshot();
    // expect(footerWithCollapse).not.toBe(footerNoCollapse);
  });

});
