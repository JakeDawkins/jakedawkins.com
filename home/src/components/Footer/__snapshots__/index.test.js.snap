exports[`Footer should exist 1`] = `
<footer
  style={null}>
  <p>
    ©2016 Jake Dawkins. Based off theme by 
    <a
      href="https://willianjusten.com.br">
      Willian Justen
    </a>
  </p>
</footer>
`;

exports[`Footer should render props correctly 1`] = `
<footer
  style={
    Object {
      "marginTop": 0
    }
  }>
  <p>
    ©2016 Jake Dawkins. Based off theme by 
    <a
      href="https://willianjusten.com.br">
      Willian Justen
    </a>
  </p>
</footer>
`;
